utils/mil_tools/nodes/online_bagger.py

```diff
@@ -319,78 +313,146 @@ def set_bag_directory(self, filename=''):
 
     def start_bagging(self, req):
         """
         Dump all data in dictionary to bags, temporarily stops streaming
         during the bagging process, resumes streaming when over.
         If bagging is already false because of an active call to this service
         """
+        result = BagResult()
         if self.streaming is False:
-            status = 'Bag Request came in while bagging, priority given to prior request'
-            rospy.logwarn(status)
-            return status
-
-        self.streaming = False
-        self.set_bag_directory(req.bag_name)
-
-        self.bagger_status = 'Total Message Count: {} \nFailed Requested Topics: \n'.format(
-            self.get_total_message_count())
-        requested_seconds = req.bag_time
-
-        selected_topics = req.topics.split()
-        for topic, (time, subscribed) in self.subscriber_list.items():
-            if not subscribed:
-                self.bagger_status = self.bagger_status + \
-                    'topic: {}\n'.format(topic)
-                continue
-            # Exclude topics that aren't in topics service argument
-            # If topics argument is empty string, include all topics
-            if len(selected_topics) > 0 and topic not in selected_topics:
-                continue
-            if len(self.topic_messages[topic]) == 0:
-                self.bagger_status = self.bagger_status + \
-                    'topic: {}\n'.format(topic)
-                continue
-
-            rospy.logdebug('message count: {:6}, topic: {}'.format(
-                self.get_topic_message_count(topic), topic))
-
-            # if no number is provided or a zero is given, bag all messages
-            types = (0, 0.0, None)
-            if req.bag_time in types:
-                bag_index = 0
-                self.bag_report = 'All messages were bagged'
-
-            # get time index the normal way
-            else:
-                # bag index will return the appropriate index to begin bagging at
-                # in order to have the requested number of seconds
-                bag_index = self.get_time_index(topic, requested_seconds)
-            rospy.logdebug('topic: {}, bag_index: {}'.format(topic, bag_index))
-            messages = 0  # message number in a given topic
-            for msgs in self.topic_messages[topic][bag_index:]:
-                messages = messages + 1
-                self.bag.write(topic, msgs[1], t=msgs[0])
-                if messages % 100 == 0:  # log every 100th topic, type and time
-                    rospy.logdebug('topic: %s, message type: %s, time: %s',
-                                   topic, type(msgs[1]), type(msgs[0]))
-
-            # empty deque when done writing to bag
-            self.topic_messages[topic].clear()
-
-        self.bag.close()
-        rospy.loginfo('bagging finished!')
+            result.status = 'Bag Request came in while bagging, priority given to prior request'
```

Comment:
Yeah that would be pretty useful (especially as a ROS community tool). Onward with online bagger development!